metadata/src/main/java/com/redhat/lightblue/metadata/Indexes.java

```diff
@@ -61,6 +63,20 @@ public boolean isEmpty() {
         return indexes.isEmpty();
     }
 
+    public boolean isCaseInsensitiveKey(Path path) {
+        return getCaseInsensitiveIndexes().stream()
+                .anyMatch(i -> i.getField().equals(path));
+
+    }
+
+    public List<IndexSortKey> getCaseInsensitiveIndexes() {
```

Comment:
That is the problem with the approach that only adds methods when needed. Mongo translator might need it that way, but another translator might need it in some other way. Instead, add methods that can be used in different situations. Index.isCaseInsensitiveKey() is more meaningful than Indexes.isCaseInsensitiveKey(). You can check if the field has any case insensitive indexes over all indexes in mongo translator.

Same goes for getCaseInsensitiveIndexes() method. It is only useful from mongo translator. Move the function there, and use Index.isCaseInsensitiveKey()